skimage/morphology/tests/test_footprints.py

```diff
@@ -279,3 +280,65 @@ def test_uneven_sequence_decomposition_warning(self, shape):
             actual = footprint_rectangle(shape, decomposition="sequence")
         assert_stacklevel(record)
         assert_equal(actual, desired)
+
+
+@pytest.mark.parametrize("width", [0, 1, 2, 3, 4, 5, 33, 103])
+@pytest.mark.parametrize("height", [0, 1, 2, 3, 4, 5, 33, 103])
+def test_footprint_ellipse_backwards_compatibility(width, height):
+    with pytest.warns(FutureWarning, match="`ellipse` is deprecated"):
+        old = ski.morphology.ellipse(width, height)
+    new = ski.morphology.footprint_ellipse(
+        shape=(height * 2 + 1, width * 2 + 1), grow=1, border=False
+    )
+    np.testing.assert_equal(old, new)
+
+
+@pytest.mark.parametrize("width", [0, 1, 2, 3, 4, 5, 33, 103])
+@pytest.mark.parametrize("height", [0, 1, 2, 3, 4, 5, 33, 103])
+def test_footprint_ellipse_backwards_compatibility_decompose(width, height):
+    with pytest.warns(FutureWarning, match="`ellipse` is deprecated"):
+        old = ski.morphology.ellipse(width, height, decomposition="crosses")
+    new = ski.morphology.footprint_ellipse(
+        shape=(height * 2 + 1, width * 2 + 1),
+        grow=1,
+        border=False,
+        decomposition="crosses",
+    )
+    np.testing.assert_equal(old, new)
+
+
+@pytest.mark.parametrize("radius", [0, 1, 2, 3, 4, 5, 33])
+def test_footprint_disk_backwards_compatibility(radius):
+    with pytest.warns(FutureWarning, match="`disk` is deprecated"):
+        old = ski.morphology.disk(radius)
+    new = ski.morphology.footprint_ellipse(shape=(radius * 2 + 1,) * 2, grow=0.0001)
+    np.testing.assert_equal(old, new)
+
+
+@pytest.mark.parametrize("radius", [0, 1, 2, 3, 4, 5, 33])
+@pytest.mark.parametrize("decomposition", ["sequence", "crosses"])
+def test_footprint_disk_backwards_compatibility_decompose(radius, decomposition):
+    with pytest.warns(FutureWarning, match="`disk` is deprecated"):
+        old = ski.morphology.disk(radius, decomposition=decomposition)
+    new = ski.morphology.footprint_ellipse(
+        shape=(radius * 2 + 1,) * 2, grow=0.0001, decomposition=decomposition
+    )
+    np.testing.assert_equal(old, new)
+
+
+@pytest.mark.parametrize("radius", [0, 1, 2, 3, 4, 5, 33])
+def test_footprint_ball_backwards_compatibility(radius):
+    with pytest.warns(FutureWarning, match="`ball` is deprecated"):
+        old = ski.morphology.ball(radius)
+    new = ski.morphology.footprint_ellipse(shape=(radius * 2 + 1,) * 3, grow=0.0001)
```

Comment:
Note that `grow=0.0001` is necessary to pass this test for `radius=33`. From https://github.com/scikit-image/scikit-image/pull/7628/commits/220f453b1582cf67f4d217246d47d568ea9adaa5

> Unfortunately, there seem to be slight differences when trying to reproduce `disk` and `ball` shaped footprints with `footprint_ellipse`. Right now I'm guessing that this might be due to floating point calculations.
>
> `disk` and `ball` construct the equation in a different way, than `footprint_ellipse` which supports N dimensions.

Since we deprecate but don't plan on removing `ball`, `disk` and `ellipse` until `skimage2`, I think this is fine. Though, ideally, we can figure out why this small correction is necessary. :thinking: